### ISSUE TITLE
HAX constructed deck less than 60 cards minimum

### ISSUE THREAD

**Issue by BillyG-of-xmage-fame** [magefree/mage#5579]
It is possible to have a constructed deck with less than 60 cards minimum in games 2 and 3 after sideboard.

What you do is sideboard out cards below the 60 minimum stated in the mtg rules, and wait for the sideboard timer to run out.

I tried to leave just 7 cards maindeck, but it defaulted to the 'limited' 40 card minimum, which means code exists to check that a timed out sideboarding stage still creates a deck with a minimum number of cards in it, but that minimum is 40 (as it should be in limited) not 60 like its supposed to be in legacy constructed.

this is 100% repeatable and the abuse is deck thinning to 40 cards is possible in sideboard games, completely ruining the game.

Screenshots of discovery:
https://imgur.com/a/A8ALny1

screenshots of test to identify limits of bug: 
https://imgur.com/a/f2YVYFT

Please fix immediately, i spammed chat about it so everyone knows, it is potentially gamebreaking to thin the deck to 40 cards in a 60 card or more format

**Comment by BillyG-of-xmage-fame** [comment]
Discovery:

![](https://i.imgur.com/88N437x.jpg)

![](https://i.imgur.com/fQrZVbR.jpg)


Testing:
![](https://i.imgur.com/ZIjcRnQ.jpg)

![](https://i.imgur.com/28m6ZWv.jpg)

**Comment by ketsuban** [comment]
I got the scent of a hardcoded value from this bug report, so I searched for instances of the integer 40 and found [this method on the `MatchPlayer` class](https://github.com/magefree/mage/blob/master/Mage/src/main/java/mage/game/match/MatchPlayer.java#L81-L89).

```java
public Deck generateDeck() {
    //TODO: improve this
    while (deck.getCards().size() < 40 && !deck.getSideboard().isEmpty()) {
        Card card = deck.getSideboard().iterator().next();
        deck.getCards().add(card);
        deck.getSideboard().remove(card);
    }
    return deck;
}
```

It's called by [this method on the `TableController` class](https://github.com/magefree/mage/blob/master/Mage.Server/src/main/java/mage/server/TableController.java#L876-L882), which matches the OP's description of the bug occurring when you allow the sideboard timer to expire without submitting - the system takes what you already did, moves cards from your sideboard to your deck until either your sideboard is empty or your deck has 40 cards in it, then submits that.

**Comment by JayDi85** [comment]
Fixed via de4befb9c26406b82059f19f1bcffbb6959bcc6a
